internal/upgradecheck/http.go

```diff
@@ -136,8 +136,12 @@ func checkForUpgrades(ctx context.Context, versionString string, backoff wait.Ba
 		err = fmt.Errorf("received StatusCode %d", status)
 	}
 
+	if len(req.Header[clientHeader]) > 0 {
+		header = req.Header[clientHeader][0]
+	}
```

Comment:
⛏️ `http.Header` has methods for this kind of thing.

```suggestion
	header = req.Header.Get(clientHeader)
```

Reply:
:+1: updated